src/daemon/rpmostreed-transaction-types.c

```diff
@@ -131,41 +133,63 @@ apply_revision_override (RpmostreedTransaction    *transaction,
                          GCancellable             *cancellable,
                          GError                  **error)
 {
+  RpmOstreeRefspecType refspectype;
+  rpmostree_origin_classify_refspec (origin, &refspectype, NULL);
+
   g_autofree char *checksum = NULL;
   g_autofree char *version = NULL;
-
-  if (!rpmostreed_parse_revision (revision,
-                                  &checksum,
-                                  &version,
-                                  error))
+  if (!rpmostreed_parse_revision (revision, &checksum, &version, error))
     return FALSE;
 
   if (version != NULL)
     {
-      rpmostreed_transaction_emit_message_printf (transaction,
-                                                  "Resolving version '%s'",
-                                                  version);
-
-      if (!rpmostreed_repo_lookup_version (repo, rpmostree_origin_get_refspec (origin),
-                                           version, progress,
-                                           cancellable, &checksum, error))
-        return FALSE;
+      switch (refspectype)
+        {
+        case RPMOSTREE_REFSPEC_TYPE_OSTREE:
+          {
+            /* Perhaps down the line we'll drive history traversal into libostree */
+            rpmostreed_transaction_emit_message_printf (transaction,
+                                                        "Resolving version '%s'",
+                                                        version);
+
+            if (!rpmostreed_repo_lookup_version (repo, rpmostree_origin_get_refspec (origin),
+                                                 version, progress,
+                                                 cancellable, &checksum, error))
+              return FALSE;
+
+            rpmostree_origin_set_override_commit (origin, checksum, version);
+          }
+          break;
+        case RPMOSTREE_REFSPEC_TYPE_ROJIG:
+          /* This case we'll look up later */
+          rpmostree_origin_set_jigdo_version (origin, version);
+          break;
+        }
     }
   else
     {
       g_assert (checksum != NULL);
 
-      rpmostreed_transaction_emit_message_printf (transaction,
-                                                  "Validating checksum '%s'",
-                                                  checksum);
+      switch (refspectype)
+        {
+        case RPMOSTREE_REFSPEC_TYPE_OSTREE:
+          rpmostreed_transaction_emit_message_printf (transaction,
+                                                      "Validating checksum '%s'",
+                                                      checksum);
+          if (!rpmostreed_repo_lookup_checksum (repo, rpmostree_origin_get_refspec (origin),
+                                                checksum, progress, cancellable, error))
+            return FALSE;
+          break;
+        case RPMOSTREE_REFSPEC_TYPE_ROJIG:
+          /* For now we skip validation here, if there's an error we'll see it later
```

Comment:
Let's just fail earlier here and give the same "not implemented" message that we would later in the upgrader?

Reply:
Seems easier to have just one place to change later?  I actually am not sure why in the `ostree://` case we do a pull-before-we-pull anyways.